demos/interactive_face_detection_demo_gapi/main.cpp

```diff
@@ -0,0 +1,582 @@
+            fileNameNoExt(FLAGS_m_lm) + ".bin",// path to weights
+            FLAGS_d_lm                         // device to use
+        }.cfgOutputLayers({ "align_fc3" });
+
+        auto emo_net = cv::gapi::ie::Params<Emotions> {
+            FLAGS_m_em,                         // path to model
+            fileNameNoExt(FLAGS_m_em) + ".bin", // path to weights
+            FLAGS_d_em                          // device to use
+        };
+
+        // Form a kernel package (including an OpenCV-based implementation of our
+        // post-processing) and a network package (holding our three networks).
+        auto kernels = cv::gapi::kernels<OCVPostProc>();
+        auto networks = cv::gapi::networks(det_net, age_net, hp_net, lm_net, emo_net);
+
+        cv::GStreamingCompiled stream = pipeline.compileStreaming(cv::compile_args(kernels, networks));
+
+        cv::Mat frame, ssd_res;
+        std::vector<cv::Rect> face_hub;
+        auto out_vector = cv::gout(frame, ssd_res, face_hub);
+
+        std::vector<cv::Mat> out_ages, out_genders;
+        if (!FLAGS_m_ag.empty()) out_vector += cv::gout(out_ages, out_genders);
+
+        std::vector<cv::Mat> out_y_fc, out_p_fc, out_r_fc;
+        if (!FLAGS_m_hp.empty()) out_vector += cv::gout(out_y_fc, out_p_fc, out_r_fc);
+
+        std::vector<cv::Mat> out_emotions;
+        if (!FLAGS_m_em.empty()) out_vector += cv::gout(out_emotions);
+
+        std::vector<cv::Mat> out_landmarks;
+        if (!FLAGS_m_lm.empty()) out_vector += cv::gout(out_landmarks);
+
+        Visualizer::Ptr visualizer;
+        if (!FLAGS_no_show || !FLAGS_o.empty()) {
+            visualizer = std::make_shared<Visualizer>(!FLAGS_m_ag.empty(), !FLAGS_m_em.empty(), !FLAGS_m_hp.empty(), !FLAGS_m_lm.empty());
+        } else {
+            std::cout<< "To close the application, press 'CTRL+C' here" << std::endl;
+        }
+
+        std::list<Face::Ptr> faces;
+        std::ostringstream out;
+        size_t framesCounter = 0;
+        size_t id = 0;
+        cv::VideoWriter videoWriter;
+
+        Timer timer;
+
+        slog::info << "Setting media source" << slog::endl;            
+        try {
+            if (FLAGS_i == "cam")
+                stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(0));
+            else
+                stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(FLAGS_i));
+        } catch (const std::exception& error) {
+            std::stringstream msg;
+            msg << "Can't open source {" << FLAGS_i << "}" << std::endl;
+            throw std::invalid_argument(msg.str());
+        }
+        slog::info << "Start inference " << slog::endl;
+        stream.start();
+
+        const cv::Point THROUGHPUT_METRIC_POSITION{10, 45};
+        std::unique_ptr<Presenter> presenter;
+
+        while (stream.running()) {
+            timer.start("total");
+
+            if (stream.pull(std::move(out_vector))) {
```

Comment:
This construction seems odd. If `out_vector` is moved into `stream.pull`, then its value becomes undefined, yet it's used again on the next iteration. Isn't something wrong here?

Reply:
There is overloading of `pull` method which looks like
`bool cv::gimpl::GStreamingExecutor::pull(cv::GRunArgsP &&outs)`
And there is operator `&&` overloading for `cv::GRunArgsP out vector`

Using `std::move` is relevant in this case, you can see the implementation details here
https://github.com/opencv/opencv/blob/624d532000d726734637e6d51c8e40785f45b46b/modules/gapi/src/executor/gstreamingexecutor.cpp#L1356

Reply:
Actually Roman's comment sounds reasonable.

`pull()` has been designed to be used like this: `pull(cv::gout(out1, out2, ...))` where `gout()` produces that `cv::GRunArgsP`.

Reply:
I got you
This really may seem some odd, but it works!

Should I make some overloading of a `cv::gout()` that takes a `cv::GRunArgsP` as input and returns a copy of it?

Reply:
The design of `pull` seems strange to me, TBH. If it doesn't modify the input vector, why does it accept it by rvalue reference?

Regardless of the answer to that question, I think the cleanest workaround would be to replace `std::move` with `cv::GRunArgsP` on this line.

Reply:
`std::move(out_vector)` -> `cv::GRunArgsP(out_vector)`